### PR TITLE
New interface definition

### PR DESCRIPTION
**Pull request by iulianpascalau**
- changed the outport driver interface definition
- added TODOs for fixing/adding new tests

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package notifier
+
+import "errors"
+
+// ErrNilTransactionPool signals that a nil transactions pool was provided
+var ErrNilTransactionPool = errors.New("nil transactions pool")
```
Comment (bogdan-rosianu) [inline errors.go:6]: ErrNilTransaction**s**Pool

File: eventNotifier.go [hunk 2]
```diff
@@ -117,36 +119,40 @@ func (en *eventNotifier) RevertIndexedBlock(header nodeData.HeaderHandler, body
...
 		Hash: hex.EncodeToString(headerHash),
 	}
 
 	err := en.httpClient.Post(finalizedEventsEndpoint, finalizedBlock, nil)
 	if err != nil {
-		log.Error("error while posting finalized event data", "err", err.Error())
+		return fmt.Errorf("%w in eventNotifier.FinalizedBlock while posting event data", err)
 	}
+
+	return err
```
Comment (bogdan-rosianu) [inline eventNotifier.go:139]: return nil  -> same thing, but a little more clean

File: interface.go [hunk 3]
```diff
@@ -6,13 +6,13 @@ import (
 )
 
 type Indexer interface {
-	SaveBlock(args *indexer.ArgsSaveBlockData)
+	SaveBlock(args *indexer.ArgsSaveBlockData) error
 	SaveRoundsInfo(roundsInfos []*indexer.RoundInfo)
 	SaveValidatorsPubKeys(validatorsPubKeys map[uint32][][]byte, epoch uint32)
-	RevertIndexedBlock(header coreData.HeaderHandler, body coreData.BodyHandler)
+	RevertIndexedBlock(header coreData.HeaderHandler, body coreData.BodyHandler) error
```
Comment (bogdan-rosianu) [inline interface.go:12]: shouldn't we change all the functions signatures for consistency?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: errors.go (hunk 1 above)
Comment (iulianpascalau) [reply errors.go:6]: done

File: eventNotifier.go (hunk 2 above)
Comment (iulianpascalau) [reply eventNotifier.go:139]: right, done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by mateigraura — APPROVED** [review]

**Pull request merged**